src/controller/CHIPDeviceController.cpp

```diff
@@ -503,6 +508,25 @@ CHIP_ERROR DeviceCommissioner::Init(NodeId localDeviceId, PersistentStorageDeleg
 {
     ReturnErrorOnFailure(DeviceController::Init(localDeviceId, storageDelegate, systemLayer, inetLayer));
 
+    if (mStorageDelegate != nullptr)
+    {
+        char keyIDStr[kMaxKeyIDStringSize];
+        uint16_t size  = sizeof(keyIDStr);
+        CHIP_ERROR err = CHIP_NO_ERROR;
+        PERSISTENT_KEY_OP(static_cast<uint64_t>(0), kNextAvailableKeyID, key,
```

Comment:
do we need the macro here?
this is harder to use and the macro seems to just prefix the hardcoded 0.
this seems equivalent to `GetKeyValue("StartKeyID0")` but much harder to read.

Reply:
we may also want to consider using the binary API to read data instead of using printf/scanf (maybe as a TODO as we are still in a state where our API is split between sync/async with partial implementation ... binary API is the sync version).